--- src/Makefile.am.orig	2018-04-11 23:35:59.000000000 +0300
+++ src/Makefile.am	2018-07-28 20:22:01.810369500 +0300
@@ -22,7 +22,7 @@
 SUBDIRS = common/avl common/net common/timing common/httpp $(MAYBE_THREAD)
 
 lib_LTLIBRARIES = libshout.la
-libshout_la_LDFLAGS = -version-info 5:0:2
+libshout_la_LDFLAGS = -version-info 5:0:2 -no-undefined
 
 EXTRA_DIST = codec_theora.c codec_speex.c tls.c
 noinst_HEADERS = format_ogg.h shout_private.h util.h
